onmt/inputters/inputter.py

```diff
@@ -197,23 +197,22 @@ def collect_features(fields, side="src"):
     return feats
 
 
-# min_len is misnamed
 def filter_example(ex, use_src_len=True, use_tgt_len=True,
-                   min_src_len=0, max_src_len=float('inf'),
-                   min_tgt_len=0, max_tgt_len=float('inf')):
+                   min_src_len=1, max_src_len=float('inf'),
+                   min_tgt_len=1, max_tgt_len=float('inf')):
     """
     A generalized function for filtering examples based on the length of their
     src or tgt values. Rather than being used by itself as the filter_pred
     argument to a dataset, it should be partially evaluated with everything
     specified except the value of the example.
     """
-    return (not use_src_len or min_src_len < len(ex.src) <= max_src_len) and \
-        (not use_tgt_len or min_tgt_len < len(ex.tgt) <= max_tgt_len)
+    return (not use_src_len or min_src_len <= len(ex.src) <= max_src_len) and \
+        (not use_tgt_len or min_tgt_len <= len(ex.tgt) <= max_tgt_len)
 
 
 def build_dataset(fields, data_type, src_data_iter=None, src_path=None,
                   src_dir=None, tgt_data_iter=None, tgt_path=None,
-                  src_seq_len=0, tgt_seq_len=0,
+                  src_seq_len=50, tgt_seq_len=50,
```

Comment:
1) I don't like the idea of hard coded values here
2) isn't it wrong at inference ?
I might be mistaken but 0 means no filtering.

Reply:
Answered in reverse order:

It is not wrong at inference. At inference, no filtering is applied. A maximum length of 0 does not mean that no filtering is applied, it means that everything with a length of 0 is filtered. That is obviously not something anyone should ever want.

I don't think it's fair to call the value hard-coded. Keyword arguments with default values are the standard for functions all throughout the codebase. Unless we refactor to reduce the number of keyword arguments, the choice is between values that do make sense and values that do not make sense.  As explained above, 0 does not make sense (it filters everything). 50 does make sense in that it's identical to the value specified in opts.py. Also, throughout the codebase `build_dataset` is called with these length values provided.

Reply:
I don't want to be picky but it makes more sense to set them to None (which is in fact the default behavior in the case of inference) rather than a value whose default is already set in opts.py

no clue why travis is erroring but I'll re-run it.

Reply:
Not filtering is the default at inference, but this is accomplished with `use_filter_pred`, not by setting anything to None. `use_filter_pred=False` means that all of the filtering-related arguments will be ignored no matter what you set them to.

I can look into changing how the arguments of `build_dataset` work if you like, and in the long term I agree with your proposed solution. I intended this pull request as a quick fix to an actual tangible problem I've been having in my research (the validation filtering thing) with minimal changes to the code. I only changed these default values because they did not make sense and I thought it was likely that other maintainers would misinterpret what they mean at first glance.

Reply:
What I meant is not to not filter based on the None value, I was just saying that value is None as a fact when it comes to inference.
anyway, merging as is and leaving it for later.